includes/admin/class.llms.admin.builder.php

```diff
@@ -640,76 +643,184 @@ private static function process_detachments( $data ) {
+		 *
+		 * A failure should contain an error message in a second array member:
+		 * `array( 'id' => $id, 'error' => esc_html__( 'My error message', 'my-domain' ) )`.
+		 *
+		 * @since Unknown.
+		 *
+		 * @param null|array $trash_response Denotes the trash response. See description above for details.
+		 * @param array      $res            The initial default error response which can be modified for your needs and then returned.
+		 * @param mixed      $id             The ID of the course element. Usually a WP_Post id.
+		 */
+		$custom = apply_filters( 'llms_builder_trash_custom_item', null, $res, $id );
+		if ( $custom ) {
+			return $custom;
+		}
 
-				$type = 'question_choice';
+		// Determine the element's post type.
+		$type = is_numeric( $id ) ? get_post_type( $id ) : false;
 
-			}
+		if ( $type ) {
+			$status = self::process_trash_item_post_type( $id, $type );
+		} else {
+			$status = self::process_trash_item_custom( $id );
+		}
 
-			$post_types = apply_filters( 'llms_builder_trashable_post_types', array( 'lesson', 'llms_quiz', 'llms_question', 'question_choice', 'section' ) );
-			if ( ! in_array( $type, $post_types ) ) {
-				array_push( $ret, $res );
-				continue;
-			}
+		// Error deleting.
+		if ( is_wp_error( $status ) ) {
+			$res['error'] = $status->get_error_message();
 
-			// lessons, sections, & questions passed as numeric WP Post IDs
-			if ( is_numeric( $id ) ) {
+		} elseif ( true === $status ) {
+			// Success.
+			unset( $res['error'] );
 
-				// delete sections
-				if ( in_array( $type, array( 'section', 'llms_question', 'llms_quiz' ) ) ) {
-					$stat = wp_delete_post( $id, true );
-				} else {
-					$stat = wp_trash_post( $id );
-				}
-			} else {
+		}
 
-				$split    = explode( ':', $id );
-				$question = llms_get_post( $split[0] );
-				if ( $question && is_a( $question, 'LLMS_Question' ) ) {
-					$stat = $question->delete_choice( $split[1] );
-				} else {
-					$stat = false;
-				}
-			}
+		return $res;
 
-			// both functions return false on failure
-			if ( ! $stat ) {
-				$res['error'] = sprintf( esc_html__( 'Error deleting %1$s "%2$s".', 'lifterlms' ), $type, $id );
-			} else {
-				unset( $res['error'] );
-			}
+	}
 
-			array_push( $ret, $res );
+	/**
+	 * Delete a question choice
+	 *
+	 * This method is called via the `llms_builder_trash_custom_item` filter. The filter
+	 * is applied and removed in the `LLMS_Admin_Builder::process_trash()`.
+	 *
+	 * Note that while this is a public method, it's not intended to be used outside of the
+	 *
+	 * @since  [version]
+	 *
+	 * @param  string $id Custom item ID. This should be a question choice id in the format of "${question_id}:{$choice_id}".
+	 * @return null|true|WP_Error `null` when the $id cannot be parsed into a question choice id.
+	 *                            `true` on success.
+	 *                            `WP_Error` when an error is encountered.
+	 */
+	private static function process_trash_item_custom( $id ) {
```

Comment:
Well... Okay it used to just be a filter that could be used by 3rd parties (assignments uses it to delete assignment tasks, for example).

The core used to have "question_choices" as a post type that was handled as a seperate condition and I decided since that's semantically incorrect to call a question choice a post type (since it's NOT a post type) that I'd add it as a "custom item" the same way assignments does.

I added it as a filter just like assignments and then realized that that made the code really hard to dissect and undid that in favor of making a "custom_items" method that deletes the question choices. I figured if we add additional items like this in the future it can be split into a switch statement or something.

I'll have another look at this, fix the docs, and see if there's a more semantic way to name this.